tokio/src/sync/mpsc/chan.rs

```diff
@@ -147,7 +147,7 @@ impl<T, S: Semaphore> Tx<T, S> {
         self.inner.semaphore.is_closed()
     }
 
-    pub(crate) async fn closed(&mut self) {
+    pub(crate) async fn closed(&self) {
         use std::future::Future;
         use std::pin::Pin;
         use std::task::Poll;
```

Comment:
I don't know enough about the internals of channels to know whether it relies on the exclusivity of a mutable reference. However if so, we do not have any loom tests that would catch it, because such a test would not compile before this change was applied.

I would like to see a loom test that calls `send` and `closed` on the same `Arc<Sender>` from multiple threads concurrently.

Reply:
@Darksonn Good point! I will verify that.

Reply:
Seems these are simply awaiting on a `Notify` so there should not be much of a problem, but still there is the loom test

Reply:
I would assume the `&mut self` requirement (if there was one) would come from the `self.closed` call but that seems to take `&self` so this should be fine.